### ISSUE TITLE
New SILVA databases?

### ISSUE THREAD

**Issue by adityabandla** [sortmerna/sortmerna#177]
Any plans to update the rRNA databases to the current version of SILVA?

**Comment by ekopylova** [comment]
Sure, by end of the month they will be ready and included in next releases.

**Comment by elarsonneur** [comment]
Nice good news.

Could you simultaneously specify indications to generate your databases in your README file please,
for the step 1 (ie arb options/filters to export fasta files by phylum and by rRNA sequence type) ?  
https://github.com/biocore/sortmerna/blob/master/rRNA_databases/README.txt

I would like to generate the 16S bacteria database clustered with higher identity percentage (97, 98, 99).

Thanks,
Elise

**Comment by elarsonneur** [comment]
Here are my questions about some steps from your README file 
https://github.com/biocore/sortmerna/blob/master/rRNA_databases/README.txt .


STEP 1 (ARB)
-------------
For example, to extract the FASTA file for 16S bacteria, did you proceed like this ?

`$ arb SSURef_NR99_119_SILVA_14_07_14_opt.arb `

Then from the arb GUI:

```
1. SEARCH
Click the 'Search' button
Search filters :
  tax_embl = Bacteria*
  and
  ARB_color != 1 (this removes low quality reads and chimeras)
  and
  product = 16S*
Click 'Search'
Click the 'Mark Listed Unmark Rest' button
Close the 'Search and Query' box

2. EXPORT
File/Export to external format
Export options:
   Export: marked
   Filter: none
   Compress: all_gaps (to get complete sequences) (or no or vertical_gaps) ?
   Select a format: fasta.eft (or other fasta template ?)
   Choose an output file name: <PATH>/bacteria_16S_119 (or other name ?)
Go
```

To extract the FASTA file for 23S bacteria, same steps but choose `product=23S*` in search filters ?


STEP 3 (HMMER_CLEAN)
-------------------------
- Can you provide scripts `remove_partial_seqs.py` and `filter_fasta.py`
used in your script `hmmer_clean.sh` please ?

- How did you build hmm files (eg `bac_ssu.hmm`, `bac_tsu.hmm`, `bac_ssu.hmm`, etc) 
or where did they come from ?


STEP 4 (GENERATE_DB)
-----------------------
What files `RF00001-T.fasta.txt` and `RF00002-T.fasta.txt` are ?

**Comment by adityabandla** [comment]
> Sure, by end of the month they will be ready and included in next releases.

Are these available now?

**Comment by ekopylova** [comment]
Sorry, not yet. I'll make it a priority to put them together for next week. Please check back again then.

**Comment by adityabandla** [comment]
> Sorry, not yet. I'll make it a priority to put them together for next week. Please check back again then.
> […](#)
> On Jan 4, 2019, 16:40, at 16:40, Aditya Bandla ***@***.***> wrote: > Sure, by end of the month they will be ready and included in next releases. Are these available now? -- You are receiving this because you commented. Reply to this email directly or view it on GitHub: [#177 (comment)](https://github.com/biocore/sortmerna/issues/177#issuecomment-451479167)

Many thanks! Appreciate the quick response

**Comment by adityabandla** [comment]
Any updates?

**Comment by ekopylova** [comment]
Yes, databases will be ready for Friday for review. Then testing and merging to master will be done next week. I can provide you with the prelim versions on Friday for your internal testing, please e-mail me if you need these then.

⁣Thanks,
Jenya

**Comment by ekopylova** [comment]
Hello,

I've re-written most steps of generating new databases and will upload them shortly. For example, we won't be using ARB files directly, rather the [exported FASTA files](https://www.arb-silva.de/no_cache/download/archive/current/Exports/) already provided on the SILVA website. These can be easily parsed to extract sequences by kindgom. In addition, many of the custom scripts to filter out partial sequences have been re-written into one python script for easier use.

Regarding your question on HMM files, they are distributed in the [Meta-RNA package](http://weizhong-lab.ucsd.edu/meta_rna/). The RF* files are RFAM databases for 5S an 5.8S.

Jenya

**Comment by adityabandla** [comment]
> Yes, databases will be ready for Friday for review. Then testing and merging to master will be done next week. I can provide you with the prelim versions on Friday for your internal testing, please e-mail me if you need these then. ⁣Thanks, Jenya
> […](#)
> On Jan 9, 2019, 15:31, at 15:31, Aditya Bandla ***@***.***> wrote: Any updates? -- You are receiving this because you commented. Reply to this email directly or view it on GitHub: [#177 (comment)](https://github.com/biocore/sortmerna/issues/177#issuecomment-452716109)

Thanks Jenya. Would be happy to test them out. My email ID is adityabandla [at] gmail [dot] com

**Comment by elarsonneur** [comment]
@ekopylova 
Thanks for your quick responses.
Yes it is a good idea to start directly from SILVA fasta files, 
rather than from ARB files which require using ARB GUI.

When will your new database be available on master branch ?

**Comment by adityabandla** [comment]
@ekopylova In case its taking long, I am willing to help generate the files and share them for review, but I would need the new steps

**Comment by ekopylova** [comment]
Thanks, that would be very helpful indeed.

Most of the steps have already been done. I have all the data on DNAnexus, could you create an account (it's free), provide me with your username and I'll add you to a new project?

Jenya

**Comment by elarsonneur** [comment]
Hi @ekopylova 
Are new Silva databases now publicly available ?
Thanks
Elise

**Comment by ekopylova** [comment]
Hello,

Not yet. 5.8S/5S are showing many shared partial sequences (e.g. 5.8S and 28S in the same sequence), and we are investigating the reasoning prior to running final clustering on all databases. I can look at this during the week and finalize the results.

New databases have not been a priority primarily because they should not recover significantly more rRNA than the current versions. For example using the entire Silva database for 16S (millions of seqs) marginally recovers more sequences than SortMeRNA's representative 16S database (thousands of seqs), for the same Silva version. The idea behind SortMeRNA searches is similar to HMMs, where new representative databases are not required at each new release of a parent database for rRNA.

Is there another reason you require new  version databases? For obtaining high scoring sequence alignments, it would make sense to use the full Silva database anyway.

Jenya

**Comment by GauravAmit** [comment]
Have you updated it yet? I am interested in making my own database from SILVA. Can you share the new protocols in regard to generating our own database from SILVA?

**Comment by MDSharma** [comment]
> 
> Hello, Not yet. 5.8S/5S are showing many shared partial sequences (e.g. 5.8S and 28S in the same sequence), and we are investigating the reasoning prior to running final clustering on all databases. I can look at this during the week and finalize the results. New databases have not been a priority primarily because they should not recover significantly more rRNA than the current versions. For example using the entire Silva database for 16S (millions of seqs) marginally recovers more sequences than SortMeRNA's representative 16S database (thousands of seqs), for the same Silva version. The idea behind SortMeRNA searches is similar to HMMs, where new representative databases are not required at each new release of a parent database for rRNA. Is there another reason you require new  version databases? For obtaining high scoring sequence alignments, it would make sense to use the full Silva database anyway. Jenya
> […](#)
> On Mar 18, 2019, 17:22, at 17:22, Elise Larsonneur ***@***.***> wrote: Hi @ekopylova Are new Silva databases now publicly available ? Thanks Elise -- You are receiving this because you were mentioned. Reply to this email directly or view it on GitHub: [#177 (comment)](https://github.com/biocore/sortmerna/issues/177#issuecomment-473987716)

Hi Jenya,

Just wondering if this was resolved at all.

Cheers,
MD

**Comment by ekopylova** [comment]
Yes, we do have the new databases. Testing them now.

**Comment by 1291987565** [comment]
Can you provide scripts remove_partial_seqs.py and filter_fasta.py
used in your script hmmer_clean.sh please ?

How did you build hmm files (eg bac_ssu.hmm, bac_tsu.hmm, bac_ssu.hmm, etc)
or where did they come from ?

**Comment by 1291987565** [comment]
I want to use the latest version of Silva to build a database

**Comment by ekopylova** [comment]
Hello!

The new databases with the latest SILVA version have been built and tested. We will provide an official link to download them within a few weeks. However, you can email me to my gmail address (jenya.kopylov) and I will provide you the link now.

FYI, with the latest SILVA version, we simplified the method to build reduced databases, which mostly included clustering kingdoms at lower % IDs and joining the clustered files. Therefore, the scripts you mention are not needed.

Jenya

**Comment by 1291987565** [comment]
I just sent you an email. Please have a look! Thanks a lot!

**Comment by elarsonneur** [comment]
Hello,

The last archive of databases is here ? :
https://github.com/biocore/sortmerna/releases/download/v4.2.0/database.tar.gz

Can you provide a README file with it ? What is the difference between default_db and sensitive_db ?

Elise

**Comment by stellaartois15** [comment]
Hello @elarsonneur ,

The differences between the two are the clustering thresholds, which are:

SMR v4.3 default db - Clustering: bac-16S 90%, 5S & 5.8S seeds, rest 95%
SMR v4.3 sensitive db - Clustering: all 97%

Jenya

**Comment by AryLech** [comment]
Hello, 

What is the difference between the four databases files?

smr_v4.3_sensituve_db_rfam_seeds.fasta
smr_v4.3_sensitive_db.fasta
smr_v4.3_fast_db.fasta
smr_v4.3_default_db.fasta

**Comment by ekopylova** [comment]
Hello,

smr_v4.3_sensitive_db_rfam_seeds.fasta -> all 97%, except 5S & 5.8S (RFAM database) which includes the full seed database sequences (benchmark accuracy: 99.907%)
smr_v4.3_sensitive_db.fasta -> all 97% (benchmark accuracy: 99.907%)
smr_v4.3_fast_db.fasta -> bac-16S 85%, 5S & 5.8S seeds, rest 90% (benchmark accuracy: 99.888%)
smr_v4.3_default_db.fasta -> bac-16S 90%, 5S & 5.8S seeds, rest 95% (benchmark accuracy: 99.899%)

We suggest to use the ```smr_v4.3_default_db.fasta```.

Thanks!
Jenya

**Comment by shuqili** [comment]
Hello,

I have similar questions for updating the SILVA database to 138.1 for sormerna version 2.1b installed via conda on a MacOS. I currently only have MacOS system to work with, so the latest available version for me could only be 2.1b. 

Because the prepackaged rRNA-database is developed based on SILVA-119, I was trying to customize the SILVA-138.1 database for better output. Could you provide the files for SILVA-138.1 database please? Or could you kindly provide instructions on how I can generate my own database based on SILVA-138.1? 

Thank you so much in advance!

Best Wishes,
Shuqi

**Comment by cmith44** [comment]
Closing this issue, for all discussion on the reference databases, see #329.
